pkg/registry/apps/daemonset/storage/storage_test.go

```diff
@@ -193,4 +197,42 @@ func TestShortNames(t *testing.T) {
 	registrytest.AssertShortNames(t, storage, expected)
 }
 
-// TODO TestUpdateStatus
+func TestUpdateStatus(t *testing.T) {
+	storage, statusStorage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.Store.DestroyFunc()
+	ctx := genericapirequest.NewDefaultContext()
+
+	key, _ := storage.KeyFunc(ctx, "foo")
+	dsStart := newValidDaemonSet()
+	err := storage.Storage.Create(ctx, key, dsStart, nil, 0)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	dsIn := &extensions.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
it's the same as that of `dsStart`, right?

maybe just `ObjectMeta: dsStart.ObjectMeta`.

Reply:
I think it is still `ObjectMeta: metav1.ObjectMeta` to stay the same with `ObjectMeta` in `extensions.DaemonSet`